internal/credentials/spiffe_test.go

```diff
@@ -178,3 +181,25 @@ func (s) TestSPIFFEIDFromState(t *testing.T) {
 		})
 	}
 }
+
+func (s) TestSPIFFEIDFromCert(t *testing.T) {
+	data, err := ioutil.ReadFile(testdata.Path("x509/spiffe_cert.pem"))
+	if err != nil {
+		t.Fatalf("Failed to load the test credentials: %v", err)
+	}
+	block, rest := pem.Decode(data)
+	if len(rest) != 0 {
+		t.Fatalf("Failed to parse the certificate: invalid format")
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("Failed to load the test credentials: %v", err)
+	}
+	uri := SPIFFEIDFromCert(cert)
+	if uri == nil {
+		t.Fatalf("SPIFFE ID is nil")
+	}
+	if uri.String() != "spiffe://foo.bar.com/client/workload/1" {
+		t.Fatalf("SPIFFE ID not expected, got %s, want %s", uri.String(), "spiffe://foo.bar.com/client/workload/1")
+	}
+}
```

Comment:
What about test cases for the other conditions checked by `SPIFFEIDFromCert`. I'm ok adding a TODO in this PR and adding those later.

Reply:
Yeah, Sure. I added some tests for the failure case.